plugins/out_stackdriver/stackdriver.c

```diff
@@ -277,6 +277,170 @@ static char *get_google_token(struct flb_stackdriver *ctx)
     return ctx->o->access_token;
 }
 
+static bool validate_msgpack_unpacked_data(msgpack_object root)
+{
+    return root.type == MSGPACK_OBJECT_ARRAY &&
+           root.via.array.size == 2 &&
+           root.via.array.ptr[1].type == MSGPACK_OBJECT_MAP;
+}
+
+static char *get_str_value_from_msgpack_map(msgpack_object_map map, 
+                                            const char *key)
+{
+    int i;
+    int key_size;
+    msgpack_object k;
+    msgpack_object v;
+    char *ptr = NULL;
+
+    key_size = strlen(key);
+    for (i = 0; i < map.size; i++) {
+        k = map.ptr[i].key;
+        v = map.ptr[i].val;
+
+        if (k.type != MSGPACK_OBJECT_STR) {
+            continue;
+        }
+
+        if (k.via.str.size == key_size && 
+            strncmp(key, (char *) k.via.str.ptr, k.via.str.size) == 0) {
+            /* make sure to free it after use */ 
+            ptr =  flb_strndup(v.via.str.ptr, v.via.str.size);
```

Comment:
use flb_sds_create_len() instead, mostly because in the caller you want to retrieve the string length, so having a flb_sds_t you can do something like flb_sds_len() which knows before hand the string length.

Reply:
updated